### DISCUSSION TITLE
How do you test upload button and table using user fixture?

### DISCUSSION THREAD

**Discussion by zx000097** [zauberzeug/nicegui#3616] [Q&A] [answered]
### Question

For instance, I have an upload button to upload a csv, then I will load a table based on the csv.

How do I test this using the user fixture? The document has somewhat limited content. 

``` ui.upload(on_upload=_handle_file_upload).mark("upload-csv") ``` 

   // refresh my table in the handle function

Currently I have: 
``` user.find("upload-csv").click()...```
What should I use to upload a test csv, and check the table has the correct row, column etc?

**Comment by rodja** [comment, marked as answer]
Thanks for reaching out, @zx000097. I created pull request #3635 which hopefully will be released very soon. It adds some documentation and shows your scenario as an example.

**Reply by rodja** [reply]
The documentation can now be seen at https://nicegui.io/documentation/user#complex_elements.
